src/firefly/js/ui/ImageSelect.jsx

```diff
@@ -0,0 +1,248 @@
+                <FilterSelect {...{onChange, type:'waveBand', dataList: waveBands}}/>
+            </CollapsiblePanel>
+
+        </div>
+    );
+}
+
+/**
+ *
+ * @param {function} onChange
+ * @param {string}   type
+ * @param {object[]} dataList
+ * @param {string}   dataList.name  ID
+ * @param {string}   dataList.label desc
+ * @param {number}   dataList.count
+ */
+class FilterSelect extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {showExpanded: false};
+    }
+
+    render() {
+        // eslint-disable-next-line
+        const {type, dataList, onChange, maxShown=3} = this.props;
+        const {showExpanded} = this.state;
+        const fieldKey= `Filter_${type}`;
+        const options = toFilterOptions(dataList);
+        const hasMore = maxShown < options.length && !showExpanded;
+        const dispOptions = showExpanded ? options : options.slice(0, maxShown);
+        const hasLess = dispOptions.length > maxShown;
+
+        return (
+            <div className='FilterPanel__item--std'>
+                <CheckboxGroupInputField
+                    key={fieldKey}
+                    fieldKey={fieldKey}
+                    initialState={{
+                                            value: '',   // workaround for _all_ for now
+                                            tooltip: 'Please select some boxes',
+                                            label : '' }}
+                    options={dispOptions}
+                    alignment='vertical'
+                    labelWidth={35}
+                    wrapperStyle={{whiteSpace: 'nowrap'}}
+                />
+
+                { hasMore && <a className='ff-href' style={{paddingLeft: 20, fontWeight: 'bold'}}
+                                onClick={() => this.setState({showExpanded: true})}>more</a>
+                }
+                { hasLess && <a className='ff-href' style={{paddingLeft: 20, fontWeight: 'bold'}}
+                                onClick={() => this.setState({showExpanded: false})}>less</a>
+                }
+            </div>
+        );
+    }
+}
+
+// save in case we want to do it this way
+// eslint-disable-next-line
+function FilterSelectExpanded({selections, onClose}) {
+    return (
+        <div className='FilterPanel__item--exp'>
+            <div style={{width: '100%', height: 15, backgroundColor: '#f3f3f3'}}>
+                <div style={{float: 'right', top: 1}}
+                     className='btn-close'
+                     title='Close popup'
+                     onClick={onClose}
+                />
+            </div>
+            <div style={{padding: '0 10px'}}>
+                {selections}
+            </div>
+        </div>
+
+    );
+}
+
+const toFilterOptions = (a) => a.map ( (d) => ({label: `${d.label}  (${d.count})`, value: d.name}));
+const toCatSummary = (master, key, desc) => Object.entries(countBy(master, (d) => `${get(d, key)};${get(d, desc)}`))
```

Comment:
This looks awesomely generic! 
You might know already and this is still in progress as we define the master table but the count (number that appears in parenthesis on the right-hand side of the mission label from the filter) should be the sum of the number of dataset each mission contains, not the number of bands or instruments. Example: in the current state of the json you are using, 2mass should have currently '2', because only LGA and AllSky surveys are present. Right?